src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java

```diff
@@ -934,6 +978,112 @@ public static CodonVariation createModification( final int codonId, final int co
+        private int insCount;
+        private int delCount;
+        private int subCount;
+
+        // start a new group with a first variation
+        public CodonVariationGroup( final int[] refCodonValues, final CodonVariation codonVariation ) {
+            this.refCodonValues = refCodonValues;
+            altCalls = new StringBuilder();
+            switch ( codonVariation.getVariationType() ) {
+                case FRAMESHIFT:
+                    isFrameShift = true;
+                    break;
+                case INSERTION:
+                    insCount += 1;
+                    altCalls.append(codonTranslation.charAt(codonVariation.getCodonValue()));
+                    break;
+                case DELETION:
+                    delCount += 1;
+                    break;
+                case MODIFICATION:
+                    subCount += 1;
+                    altCalls.append(codonTranslation.charAt(codonVariation.getCodonValue()));
+                    break;
+            }
+            startingCodon = endingCodon = codonVariation.getCodonId();
+        }
+
+        // attempt to add more variations
+        // returns false if the presented variation cannot be added to the group
+        public boolean addVariation( final CodonVariation codonVariation ) {
+            final int codonId = codonVariation.getCodonId();
+            if ( codonId > endingCodon + 1 && !isFrameShift ) return false;
+            switch ( codonVariation.getVariationType() ) {
+                case FRAMESHIFT:
+                    return false;
+                case INSERTION:
+                    insCount += 1;
+                    altCalls.append(codonTranslation.charAt(codonVariation.getCodonValue()));
+                    if ( isFrameShift && isEmpty() ) startingCodon = codonId;
+                    break;
+                case DELETION:
+                    delCount += 1;
+                    break;
+                case MODIFICATION:
+                    final char aa = codonTranslation.charAt(codonVariation.getCodonValue());
+                    if ( aa == codonTranslation.charAt(refCodonValues[codonId]) ) {
+                        if ( isFrameShift ) {
+                            if ( !isEmpty() ) altCalls.append(aa);
+                            break;
+                        }
+                        return false;
+                    }
+                    if ( isFrameShift && isEmpty() ) startingCodon = codonId;
+                    subCount += 1;
+                    altCalls.append(aa);
+                    break;
+            }
+            endingCodon = codonId;
+            return true;
+        }
+
+        // sometimes when starting with a frame shift you end up with nothing because the variants are all synonymous
+        public boolean isEmpty() { return subCount + insCount + delCount == 0; }
+
+        // translate the group into standard nomenclature
+        @Override public String toString() {
```

Comment:
may I suggest a different name for this method?
`tostring()` to me is almost always for debugging.
This method is clearly doing very important tasks.

What about `toStandardNomanclature()`?